dataactbroker/handlers/fileHandler.py

```diff
@@ -890,6 +888,131 @@ def generateFile(self):
         # Return same response as check generation route
         return self.checkGeneration(submission_id, file_type)
 
+    def generate_detached_file(self):
+        """ Start a file generation job for the specified file type """
+        _debug_logger.debug('Starting D file generation')
+        request_dict = RequestDictionary(self.request)
+
+        if not(request_dict.exists("file_type") and request_dict.exists("cgac_code")):
+            raise ResponseException("Generate detached file route requires file_type and cgac_code", StatusCode.CLIENT_ERROR)
+
+        # Get file_type (D1 | D2) and cgac_code from the request dict
+        file_type = request_dict.getValue("file_type")
+        cgac_code = request_dict.getValue("cgac_code")
+
+        if file_type in ['D1', 'D2']:
+            if not (request_dict.exists("start") and request_dict.exists("end")):
+                raise ResponseException("Generate detached file route requires 'start' and 'end' for generating D1 and D2 files",
+                                        StatusCode.CLIENT_ERROR)
+
+            # Populate start and end dates, these should be provided in
+            # MM/DD/YYYY format, using calendar year (not fiscal year)
+            start_date = request_dict.getValue("start")
+            end_date = request_dict.getValue("end")
+
+            if not (StringCleaner.isDate(start_date) and StringCleaner.isDate(end_date)):
+                raise ResponseException("Start or end date cannot be parsed into a date", StatusCode.CLIENT_ERROR)
+
+
+        # add job info
+        sess = GlobalDB.db().session
+        user_id = LoginSession.getName(session)
+        file_type_name = FILE_TYPE_DICT_ID[FILE_TYPE_DICT_LETTER_ID[file_type]]
+        timestamped_name = s3UrlHandler.getTimestampedFilename(CONFIG_BROKER["".join([str(file_type_name),"_file_name"])])
+        if self.isLocal:
+            upload_file_name = "".join([CONFIG_BROKER['broker_files'], timestamped_name])
+        else:
+            upload_file_name = "".join([str(user_id), "/", timestamped_name])
+
+        new_job = Job(filename=upload_file_name, original_filename=timestamped_name,
+                      job_status_id=JOB_STATUS_DICT['running'], job_type_id=JOB_TYPE_DICT['file_upload'],
+                      user_id=user_id, file_type_id=FILE_TYPE_DICT_LETTER_ID[file_type], start_date=start_date,
+                      end_date=end_date)
```

Comment:
`start_date` and `end_date` are only defined if the file type is D1 or D2. Otherwise this'd explode, yes?